### PR TITLE
21003: Add attributes to account for separate gateway specific iam roles

### PR DESCRIPTION
**Pull request by mlin-aviatrix**
Changed aws_role_app and was_role_ec2 to optional and computed attributes.

### CODE REVIEW

File: aviatrix/resource_aviatrix_account.go [hunk 1]
```diff
@@ -55,15 +55,15 @@ func resourceAviatrixAccount() *schema.Resource {
...
 				Optional:    true,
-				Description: "AWS App role ARN.",
+				Description: "AWS App role ARN for gateways.",
 			},
-			"aws_role_ec2": {
+			"aws_gateway_role_ec2": {
 				Type:        schema.TypeString,
 				Optional:    true,
-				Description: "AWS EC2 role ARN.",
+				Description: "AWS App role ARN for gateways.",
```
Comment (CyrusJavan) [inline aviatrix/resource_aviatrix_account.go]: These have the exact same description?

File: aviatrix/resource_aviatrix_account.go [hunk 2]
```diff
@@ -195,23 +209,29 @@ func resourceAviatrixAccountCreate(d *schema.ResourceData, meta interface{}) err
...
-			role_ec2.WriteString(account.AwsAccountNumber)
-			role_ec2.WriteString(":role/aviatrix-role-ec2")
-			if aws_role_app := d.Get("aws_role_app").(string); aws_role_app == "" {
-				account.AwsRoleApp += role_app.String()
+		if awsIam {
+			if _, ok := d.GetOk("aws_role_app"); !ok {
+				var roleApp bytes.Buffer
+				roleApp.WriteString("arn:aws:iam::")
+				roleApp.WriteString(account.AwsAccountNumber)
+				roleApp.WriteString(":role/aviatrix-role-app")
```
Comment (CyrusJavan) [inline aviatrix/resource_aviatrix_account.go]: Nitpick, don't have to change: I know you didn't write this originally, but shouldn't this just be a fmt.Sprintf ? not sure why we use a byte buffer for such a small string

**Review by CyrusJavan — DISMISSED** [review]

File: aviatrix/resource_aviatrix_account.go (hunk 2 above)
Comment (mlin-aviatrix) [reply aviatrix/resource_aviatrix_account.go]: This makes sense. I changed it to fmt.Sprintf.

**Review by CyrusJavan — DISMISSED** [review]

**Review by CyrusJavan — APPROVED** [review]

File: aviatrix/resource_aviatrix_account.go [hunk 3]
```diff
@@ -222,23 +235,21 @@ func resourceAviatrixAccountCreate(d *schema.ResourceData, meta interface{}) err
...
+				account.AwsRoleEc2 = fmt.Sprintf("arn:aws:iam::%s:role/aviatrix-role-ec2", account.AwsAccountNumber)
 			}
 			log.Printf("[TRACE] Reading Aviatrix account aws_role_app: [%s]", d.Get("aws_role_app").(string))
 			log.Printf("[TRACE] Reading Aviatrix account aws_role_ec2: [%s]", d.Get("aws_role_ec2").(string))
+
+			_, gatewayRoleAppOk := d.GetOk("aws_gateway_role_app")
+			_, gatewayRoleEc2Ok := d.GetOk("aws_gateway_role_ec2")
+			if gatewayRoleAppOk != gatewayRoleEc2Ok {
+				return fmt.Errorf("failed to create Aviatrix account: must provide both gateway app role ARN and gateway ec2 role ARN when using separate IAM role and policy for gateways")
+			}
```
Comment (shuoliugit) [inline aviatrix/resource_aviatrix_account.go:252]: If neither of them is set, then no error will be returned, right?

**Review by zhiqiangjin — APPROVED** [review]

**Pull request merged**